src/webview/message-viewer.ts

```diff
@@ -444,6 +450,24 @@ class MessageViewerViewModel extends ViewModel {
   consumeMode = this.resolve<ConsumeMode>(() => post("GetConsumeMode"), "beginning");
   consumeModeTimestamp = this.resolve(() => post("GetConsumeModeTimestamp"), Date.now());
 
+  /** Timezone display mode for timestamps (local or UTC) */
+  timestampDisplayMode = this.signal(storage.get()?.timestampDisplayMode ?? "utc");
+
+  /** Formatted timestamp for datetime-local input (YYYY-MM-DDTHH:mm:ss.sss) */
+  consumeModeTimestampFormatted = this.derive(() => {
+    const timestamp: number | null = this.consumeModeTimestamp();
+    const displayMode = this.timestampDisplayMode();
+
+    if (!timestamp) return "";
+
+    if (displayMode === "local") {
+      return timestampToDatetimeLocal(timestamp);
+    } else {
+      // For UTC mode, use ISO string format with Z suffix to match message grid
+      return new Date(timestamp).toISOString();
```

Comment:
If `timestamp` cannot be parsed as valid date, `.toISOString()` going to throw an exception even if `new Date(invalid_thing)` doesn't

Reply:
@noeldevelops based on my understanding, yes, since switching to timestamp mode should always initially use the "now" timestamp and any changes will either be invalid (and revert to the "now" timestamp) or be valid and set `consumeModeTimestamp`